_posts/2024-11-08-navigating-the-ecosystem.markdown

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title:  "Finding your way around the CHERIoT ecosystem"
+date:   2024-11-08
+categories: rtos sonata git
+author: Phil Day
+---
+With a new set of folks in the DSbD cohort 6 just getting started with their Sonata boards I thought it might be useful to capture some of the things I've learn in the last few months about the ecosystem.
+
+### Setting up a dev environment
+The easiest and quickest way to get up and running is to use the dev container.
+I have a fairly strong aversion to installing new software on my base system, and this just requires that you have docker installed.
+Opening a repo with dev container support in VS code will ask you if you want to reopen in a container.
+Do that and in a few seconds you're in a pre-built dev environment.
+Simples, to paraphrase a meerkat.
+
+One word of warning though; cheriot and Sonata both use the ":latest" tag as the version of the docker image, so you won't automatically get updates when they are released.
```

Comment:
There is now a date tagged version of the dev container, but we haven't updated any of the repos to use it yet.   I agree that would be worth doing (at least for cherito-demos, rtos might prefer to stay on the edge I guess).   Though for now best to just warn folks  to look out for it ;-)

Reply:
It's normally fine to use a newer one.  We might want to enforce a minimum.  It would probably be annoying to keep bumping it for each new build and I don't want to make it too hard for people to downgrade.